File: example/Storage/eq.py
import sys
from pathlib import Path
home = str(Path.home())
sys.path.append(home + '/Cytnx_lib')
from cytnx import *


A = Storage(10)
print(A.dtype_str())

B = A
C = A.clone()

print(B == A) # true (share same instance)
print(B is A) # true (share same instance)

print(C == A)  # true (the same content.)
print(C is A)  # false (not share same instance)

File: example/Storage/eq.cpp
#include "cytnx.hpp"
#include <iostream>

using namespace cytnx;
int main() {
  /*
      1. Create a Storage with
      dtype =Type.Double [default],
  */
  Storage A(10);
  std::cout << A.dtype_str() << std::endl;

  Storage B = A;
  Storage C = A.clone();

  std::cout << (B == A) << std::endl;  // true (share same instance)
  std::cout << is(B, A) << std::endl;  // true (share same instance)

  std::cout << (C == A) << std::endl;  // true (the same content.)
  std::cout << is(C, A) << std::endl;  // false (not share same instance)

  return 0;
}
